test/xdotoolify-spec.js

```diff
@@ -493,6 +509,76 @@ describe('xdotoolify', function() {
       'Most recent value: {"a":1,"b":2}\n' +
       'Most recent check result: {"a":1,"b":2}\n'
     );
+  });
+
+  it('should accept Xdotoolify.arg as argument', async function() {
+    let goodFunc = Xdotoolify.setupWithPage(async (page) => { console.log('running'); return {
+      a: 1,
+      b: 2
+    }; });
+
+    let funcWithArgs = Xdotoolify.setupWithPage((page, arg) => { return arg; });
+
+    let errorMsg = 'Nothing thrown';
+
+    try {
+      await page.X
+          .checkUntil(
+            funcWithArgs,
+            Xdotoolify.arg(async (_page) => {
```

Comment:
Hmm I am having second thoughts about the name `arg` because it does not communicate what it's doing to the reader.

Maybe we can use `defer` instead e.g. 
```
.checkUntil(myFunc, defer(() => myApi.call()), x => expect(x).toBe(5))
```

Reply:
changed to defer.